internal/server/data/identity.go

```diff
@@ -70,11 +70,16 @@ func AssignIdentityToGroups(db *gorm.DB, user *models.Identity, provider *models
 			groupID = group.ID
 		}
 
-		// add user to group
-		err = db.Exec("insert into identities_groups (identity_id, group_id) values (?, ?)", user.ID, groupID).Error
-		if err != nil {
-			if !isUniqueConstraintViolation(err) {
-				return fmt.Errorf("insert: %w", err)
+		var ids []uid.ID
+		if err := db.Raw("SELECT identity_id FROM identities_groups WHERE identity_id = ? AND group_id = ?", user.ID, groupID).Scan(&ids).Error; err != nil {
```

Comment:
This is necessary because the previous behaviour executed the insert regardless if the record already exists and discarded the error on unique constraint violations. This checks before insert and only inserts the record if it doesn't already exist